src/averageAges.js

```diff
@@ -52,7 +66,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const allWomen = people.filter(person => person['sex'] === 'f');
+  const allChidren = people.filter(person => allWomen.map(woman => woman['name']).includes(person['mother']));
```

Comment:
For my mind will be better(and more performant) to use `.some` instead of `.map` + `.includes`